drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -436,10 +441,10 @@ local function create_thermostat_modes_profile(device)
   return thermostat_modes
 end
 
-local function do_configure(driver, device)
+local function match_profile(driver, device)
```

Comment:
As a blanket statement, the same concerns I mentioned in regards to the timing and order of the lifecycle events on the other two drivers apply to this driver as well.

Reply:
I made the same change here as I did for matter-sensor (moved the read request from `device_added` to `do_configure` to ensure that it occurs after `match_profile` runs for the first time.